### PR TITLE
Whitespace Blob, File, and Directory Names

### PR DESCRIPTION
**Pull request by seanmcc-msft**
[Resolves 42891](https://github.com/Azure/azure-sdk-for-net/issues/42891)

### CODE REVIEW

**Comment by seanmcc-msft** [comment]
TODO:
- Changelogs
- Blob fix
- DataLake fix
- fix test recording issue

**Comment by seanmcc-msft** [comment]
@schaabs this PR is blocked based on an issue in the test framework.  Please see the email thread "Issue with test recording framework and all whitespace blob, file, and directory names" for context.

File: sdk/storage/Azure.Storage.Files.Shares/assets.json [hunk 1]
```diff
@@ -2,5 +2,5 @@
   "AssetsRepo": "Azure/azure-sdk-assets",
   "AssetsRepoPrefixPath": "net",
   "TagPrefix": "net/storage/Azure.Storage.Files.Shares",
-  "Tag": "net/storage/Azure.Storage.Files.Shares_87d726d414"
+  "Tag": "net/storage/Azure.Storage.Files.Shares_98e25ecd8b"
```
Comment (scbedd) [inline sdk/storage/Azure.Storage.Files.Shares/assets.json:5]:
```suggestion
  "Tag": "net/storage/Azure.Storage.Files.Shares_62a85d463c"
```

Comment (scbedd) [reply sdk/storage/Azure.Storage.Files.Shares/assets.json:5]: Locally fixed recordings to unblock your fix.

**Pull request closed without merging**